File: 1421-find-numbers-with-even-number-of-digits/1421-find-numbers-with-even-number-of-digits.py
class Solution:
    def findNumbers(self, nums):
        count = 0
        for x in nums:
            if self.isEven(x):
                count += 1
        return count

    def isEven(self, num):
        num_digits = self.nod(num)

        return num_digits % 2 == 0

    def nod(self, num):
        if num < 0:
            num *= -1

        return int(math.log10(num)) + 1

File: 1421-find-numbers-with-even-number-of-digits/1421-find-numbers-with-even-number-of-digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int count = 0;
        for (int x : nums) {
            if (isEven(x)) {
                count++;
            }
        }
        return count;
    }

    boolean isEven(int num) {
        int numberOfDigits = nod(num);

        return (numberOfDigits % 2 == 0);
    }

    int nod(int num) {
        if (num < 0)
            num *= -1;

        return (int) (Math.log10(num) + 1);
    }
}